File: computer_science/data-structure/hash-table/check-voter.js
import { HashTable } from "./hash-table.js";

const voted = new HashTable();
function checkVoter(name) {
  const isVoted = voted.get(name);
  if (isVoted) {
    console.log("You already voted, you can't vote again");
  } else {
    voted.set(name, true);
    console.log("Voted!");
  }
}

File: computer_science/data-structure/hash/check-voter.py
voted = {}


def checkVoter(name):
    isVoted = voted.get(name)
    if isVoted:
        print("You already voted, you can't vote again")
    else:
        voted[name] = True
        print("Voted!")


checkVoter('Tom')
checkVoter('mike')
checkVoter('mike')
